File: backend/app/models/Resep.go
package models

import "time"

type Resep struct {
	ID           string    `json:"id"`
	PenjualanID  *string   `json:"penjualan_id"`
	PelangganID  *string   `json:"pelanggan_id"`
	DokterID     *string   `json:"dokter_id"`
	NomorResep   *string   `json:"nomor_resep"`
	TanggalResep time.Time `json:"tanggal_resep"`
	NamaPasien   *string   `json:"nama_pasien"`
	UmurPasien   *string   `json:"umur_pasien"`
	AlamatPasien *string   `json:"alamat_pasien"`
	Catatan      *string   `json:"catatan"`
	Status       string    `json:"status"`
	DibuatOleh   *string   `json:"dibuat_oleh"`
	CreatedAt    time.Time `json:"created_at"`
	UpdatedAt    time.Time `json:"updated_at"`
}

File: src/types/resep.ts
export type StatusResep = "menunggu" | "diproses" | "selesai" | "ditolak";

export interface ResepDetail {
  id: string;
  resepId: string;
  barangId?: string;
  namaBarang?: string;
  satuanId?: string;
  jumlah: number;
  aturanPakai?: string;
  instruksiRacikan?: string;
  racikan: boolean;
  catatan?: string;
}

export interface Resep {
  id: string;
  nomorResep?: string;
  pelangganId?: string;
  namaPelanggan: string;
  penjualanId?: string;
  dokterId?: string;
  namaDokter: string;
  noSipDokter?: string;
  asalPuskesmas?: string;
  tanggalResep: string;
  namaPasien?: string;
  umurPasien?: string;
  alamatPasien?: string;
  catatan?: string;
  status: StatusResep;
  createdBy?: string;
  createdAt: string;
  updatedAt: string;
  details: ResepDetail[];
}
